### PR TITLE
♻️Change custom ad request batching 

### PR DESCRIPTION
**Pull request by calebcordry**
Amp custom ads currently have a feature that allows multiple ads to be fetched on a single request. However if we want to make multiple requests to the same endpoint (eg an ad server delivers dynamic data) this same logic will cause there to be only one request to be made.

This changes the "single request" logic to only be enabled if there is a `data-slot` attribute present on the ad.

### CODE REVIEW

File: extensions/amp-ad/0.1/amp-ad-custom.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ export class AmpAdCustom extends AMP.BaseElement {
     /** @const {string} fullUrl */
     const fullUrl = this.getFullUrl_();
     // If this promise has no URL yet, create one for it.
-    if (!(fullUrl in ampCustomadXhrPromises)) {
+    if (this.slot_ === null || !(fullUrl in ampCustomadXhrPromises)) {
```
Comment (zhouyx) [inline extensions/amp-ad/0.1/amp-ad-custom.js]: if two requests share the same fullUrl, Wouldn't `ampCustomadXhrPromises[fullUrl]` be replaced by the second request.

Comment (calebcordry) [reply extensions/amp-ad/0.1/amp-ad-custom.js]: Isn't that what we would want if there is no `data-slot`?

Comment (zhouyx) [reply extensions/amp-ad/0.1/amp-ad-custom.js]: Well functional wise they are the same thing. But we don't really need to check with `ampCustomadXhrPromises` array if there's no `data-slot`. What's the point of keeping them in the array.

Comment (zhouyx) [reply extensions/amp-ad/0.1/amp-ad-custom.js]: To clarify, we don't want to keep something we won't used later to the `ampCustomadXhrPromises` map. Can we do something like
```
let promise = null;
promise = ampCustomadXhrPromises[fullUrl] || Services.xhrFor(this.win)....
if (this.slot_) {
  ampCustomadXhrPromises[fullUrl] = promise;
}
return promise.then(() => {});
```

Comment (calebcordry) [reply extensions/amp-ad/0.1/amp-ad-custom.js]: You are right, this is better. Refactored.

**Pull request merged**